lua/entities/gmod_tardis_interior/modules/sv_health.lua

```diff
@@ -0,0 +1,23 @@
+ENT:AddHook("OnTakeDamage", "Health", function(self, dmginfo)
+    local newhealth = self.exterior:GetHealth() - (dmginfo:GetDamage()/2)
+    self.exterior:ChangeHealth(newhealth)
+end)
+
+function ENT:Explode()
+    if not vFireInstalled then
+        local explode = ents.Create("env_explosion")
+        explode:SetPos( self:LocalToWorld(Vector(0,0,0)) ) //Puts the explosion where you are aiming
+        explode:SetOwner( self ) //Sets the owner of the explosion
+        explode:Spawn()
+        explode:SetKeyValue("iMagnitude","100") //Sets the magnitude of the explosion
+        explode:Fire("Explode", 0, 0 ) //Tells the explode entity to explode
+        //explode:EmitSound("weapon_AWP.Single", 400, 400 ) //Adds sound to the explosion
+    else
+        //Alternate Explosion
```

Comment:
I remember seeing this code somewhere else too, could it be moved into a common function perhaps?

Also this alternate explosion is not implemented same as the other place

Reply:
I guess it makes sense to move it somewhere else.
Alternate explosion shouldn't be necessary in exterior since VFire doesn't destroy the edict count on that one

Reply:
In VFire does it automatically create an explosion then without adding any code here or is this still a todo?

Reply:
No, I still need to add an alternative explosion effect on the interior when VFire is enabled. Parts catch on fire and each fire is an entity. Given enough parts, the engine runs out of entity slots and crashes